lib/rules/no-unused-vars.js

```diff
@@ -223,6 +223,32 @@ module.exports = {
             return false;
         }
 
+        /**
+         * Gets a list of function definitions for a specified variable.
+         * @param {Variable} variable - eslint-scope variable object.
+         * @returns {ASTNode[]} Function nodes.
+         * @private
+         */
+        function getFunctionDefinitions(variable) {
```

Comment:
I would suggest 'map' usage instead of 'forEach':
```js
const getFunctionDefinitions = variable =>
    variable.defs.map(def => {
        const { type, node } = def;
        // FunctionDeclarations
        if (type === "FunctionName") {
            return node;
        }
        // FunctionExpressions
        if (type === "Variable" && node.init &&
            (node.init.type === "FunctionExpression" || node.init.type === "ArrowFunctionExpression")) {
            return node.init;
        }
    }).filter(a => a);
```

Reply:
I wouldn't like to use both `map` and `filter` here, but let's wait maintainers' feedback